### PR TITLE
indilib: set platform to linux

### PR DESCRIPTION
**Pull request by hjones2199**
###### Motivation for this change
Support building indilib on non x86_64 architectures and make minor style/format changes to indilib.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hjones2199** [comment]
Result of `nixpkgs-review pr 118696` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>indi-3rdparty</li>
    <li>indi-full</li>
    <li>indilib</li>
    <li>kstars</li>
    <li>phd2</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118696` at d18d31cb run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>kstars</li>
  </ul>
</details>
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>indi-3rdparty</li>
    <li>indi-full</li>
    <li>indilib</li>
  </ul>
</details>

File: pkgs/development/libraries/science/astronomy/indilib/default.nix [hunk 1]
```diff
@@ -24,20 +24,11 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-W6LfrKL56K1B6srEfbNcq1MZwg7Oj8qoJkQ83ZhYhFs=";
   };
 
-  nativeBuildInputs = [
-    cmake
-  ];
+  nativeBuildInputs = [ cmake ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/astronomy/indilib/default.nix:27]:
```suggestion
  nativeBuildInputs = [
    cmake
  ];
```

File: pkgs/development/libraries/science/astronomy/indilib/default.nix [hunk 2]
```diff
@@ -24,20 +24,11 @@ stdenv.mkDerivation rec {
...
-    cfitsio
-    libusb1
-    zlib
-    boost
-    libnova
-    libjpeg
-    gsl
-    fftw
+    curl boost zlib libusb1 gsl fftw
+    libnova cfitsio libjpeg
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/astronomy/indilib/default.nix:31]:
```suggestion
    curl
    cfitsio
    libusb1
    zlib
    boost
    libnova
    libjpeg
    gsl
    fftw
```

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**